src/main/kotlin/com/exactpro/th2/common/message/MessageUtils.kt

```diff
@@ -298,37 +300,82 @@ val Message.direction
 var Message.Builder.direction
     get(): Direction = metadata.id.direction
     set(value) {
-        setMetadata(MessageMetadata.newBuilder(metadata).apply {
-            setId(MessageID.newBuilder(id).apply {
-                direction = value
-            })
-        })
+        metadataBuilder.idBuilder.direction = value
+    }
+
+val RawMessage.direction
+    get(): Direction = metadata.id.direction
+var RawMessage.Builder.direction
+    get(): Direction = metadata.id.direction
+    set(value) {
+        metadataBuilder.idBuilder.direction = value
     }
 
 val Message.sessionAlias
     get(): String = metadata.id.connectionId.sessionAlias
 var Message.Builder.sessionAlias
     get(): String = metadata.id.connectionId.sessionAlias
     set(value) {
-        setMetadata(MessageMetadata.newBuilder(metadata).apply {
-            setId(MessageID.newBuilder(id).apply {
-                setConnectionId(ConnectionID.newBuilder(connectionId).apply {
-                    sessionAlias = value
-                })
-            })
-        })
+        metadataBuilder.idBuilder.connectionIdBuilder.sessionAlias = value
+    }
+
+val RawMessage.sessionAlias
+    get(): String = metadata.id.connectionId.sessionAlias
+var RawMessage.Builder.sessionAlias
+    get(): String = metadata.id.connectionId.sessionAlias
+    set(value) {
+        metadataBuilder.idBuilder.connectionIdBuilder.sessionAlias = value
     }
 
 val Message.sequence
     get(): Long = metadata.id.sequence
 var Message.Builder.sequence
     get(): Long = metadata.id.sequence
     set(value) {
-        setMetadata(MessageMetadata.newBuilder(metadata).apply {
-            setId(MessageID.newBuilder(id).apply {
-                sequence = value
-            })
-        })
+        metadataBuilder.idBuilder.sequence = value
+    }
+
+val RawMessage.sequence
+    get(): Long = metadata.id.sequence
+var RawMessage.Builder.sequence
+    get(): Long = metadata.id.sequence
+    set(value) {
+        metadataBuilder.idBuilder.sequence = value
+    }
+
+val Message.subsequence
+    get(): List<Int> = metadata.id.subsequenceList
+var Message.Builder.subsequence
+    get(): List<Int> = metadata.id.subsequenceList
+    set(value) {
+        metadataBuilder.idBuilder.apply {
+            clearSubsequence()
+            addAllSubsequence(value)
+        }
+    }
+
+val RawMessage.subsequence
+    get(): List<Int> = metadata.id.subsequenceList
+var RawMessage.Builder.subsequence
+    get(): List<Int> = metadata.id.subsequenceList
+    set(value) {
+        metadataBuilder.idBuilder.apply {
+            clearSubsequence()
+            addAllSubsequence(value)
+        }
+    }
+
+val Message.logId: String
+    get() = "$sessionAlias:${direction.toString().toLowerCase()}:$sequence${subsequence.joinToString("") { ":$it" }}"
+
+val RawMessage.logId: String
+    get() = "$sessionAlias:${direction.toString().toLowerCase()}:$sequence${subsequence.joinToString("") { ":$it" }}"
```

Comment:
Provider will use dots to separate subsequences. I think we should use the same format
```suggestion
    get() = "$sessionAlias:${direction.toString().toLowerCase()}:$sequence${subsequence.joinToString("") { ".$it" }}"

val RawMessage.logId: String
    get() = "$sessionAlias:${direction.toString().toLowerCase()}:$sequence${subsequence.joinToString("") { ".$it" }}"
```